crates/rpc/rpc/src/trace.rs

```diff
@@ -549,6 +579,41 @@ where
         }
         traces
     }
+
+    /// Returns all storage slots accessed during transaction execution along with their access
+    /// counts.
+    pub async fn trace_block_storage_access(
+        &self,
+        block_id: BlockId,
+    ) -> Result<Option<BlockStorageAccess>, Eth::Error> {
+        let res = self
+            .eth_api()
+            .trace_block_inspector(
+                block_id,
+                None,
+                StorageInspector::default,
+                move |tx_info, inspector, _res, _, _| {
```

Comment:
this is a bit more complex because we need to track warm/cold across the entire block, so this means that we need to use the same inspector, but I'm actually not entirely sure this is possible like this

https://github.com/paradigmxyz/reth/blob/f24570844545b55f2293a3a3c475133a3f738b68/crates/rpc/rpc-eth-api/src/helpers/trace.rs#L451-L451

but ideally this should be a closure that returns a &mut StorageInspector instance or includes Arc Mutex that we then need to set after the inspector run again